### ISSUE TITLE
Implement zoneout

### ISSUE THREAD

**Issue by unnonouno** [chainer/chainer#1339]
http://arxiv.org/abs/1606.01305

**Comment by fukatani** [comment]
Hi! I interested in this issue.
Here is my idea.

1) Make upward and lateral of `link.LSTMBase` configurable.

For example,
```python
    def __init__(self, in_size, out_size,
                 lateral_init=None, upward_init=None,
                 bias_init=0, forget_bias_init=0,
                 upward=None, lateral=None):
        if upward is None:
            upward=linear.Linear(in_size, 4 * out_size, initialW=0)
        if lateral is None:
            lateral =linear.Linear(in_size, 4 * out_size, initialW=0),
                                  initialW=0, nobias=True),
        super(LSTMBase, self).__init__(
            upward=upward,
            lateral=lateral,
        )
```
Upward and lateral can be any link.
This is not only for Zoneout, but also for convolutional or any other links. (See #1288).

2) Zoneout implemented as `link`.

3)Implement ZoneoutLSTM.
Its `__init__` starts with
`super(LSTM, self).__init__ with ZoneOut(linear.Linear(in_size, 4 * out_size, initialW))`

How about this plan?
If OK, I ready to implement Zoneout.

**Comment by beam2d** [comment]
It lookg zoneout has already been implemented, but is not added to the docs.

**Comment by delta2323** [comment]
Solved via #1457 and #3957
